### PR TITLE
network: rename Neighbor.MACAddress= and also support IPv4 address

### LINKED ISSUE #13015: networkd: Neighbor lladdr might be wrong format
Sample config snipped for gre tunnel:

```
[...]
[Neighbor]
Address = 10.0.0.22
MACAddress = 172.16.2.22
```

MACAddress= only accepts a mac address (struct ether_addr).
But (e.g.) gre tunnels are expecting an ipv4 addr.

`ip n add 10.0.0.22 lladdr 172.16.2.22 dev gre1`

> 10.0.0.22 dev gre1 lladdr 172.16.2.22 PERMANENT

So MACAddress should be LLAddress (or something like that) and needs to be parsed based on the device type.

**systemd version the issue has been seen with**
current master (242.1218)

**Used distribution**
Arch Linux

Thanks.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #13015.

### CODE REVIEW

File: src/network/networkd-neighbor.c [hunk 1]
```diff
@@ -133,9 +133,15 @@ int neighbor_configure(Neighbor *neighbor, Link *link, link_netlink_message_hand
...
-                return log_error_errno(r, "Could not append NDA_LLADDR attribute: %m");
+        if (neighbor->use_mac) {
+                r = sd_netlink_message_append_data(req, NDA_LLADDR, &neighbor->lladdr.mac, sizeof(neighbor->lladdr.mac));
+                if (r < 0)
+                        return log_error_errno(r, "Could not append NDA_LLADDR attribute: %m");
+        } else {
+                r = sd_netlink_message_append_data(req, NDA_LLADDR, &neighbor->lladdr.ip.in, sizeof(neighbor->lladdr.ip.in));
+                if (r < 0)
+                        return log_error_errno(r, "Could not append NDA_LLADDR attribute: %m");
+        }
```
Comment (poettering) [inline src/network/networkd-neighbor.c]: you could use the failure check and error message for both if branches. i.e.

```c
if (neighbor->use_mac) 
        r = sd_netlink_message_append_data(…
else
        r = sd_netlink_message_append_data(…
if (r < 0) 
        return …
```

File: src/network/networkd-neighbor.h [hunk 2]
```diff
@@ -22,8 +22,12 @@ struct Neighbor {
...
         int family;
         union in_addr_union in_addr;
-        bool mac_configured;
-        struct ether_addr mac;
+        union {
+                struct ether_addr mac;
+                union in_addr_union ip;
+        } lladdr;
+        bool use_mac:1;
+        bool use_ip:1;
```
Comment (poettering) [inline src/network/networkd-neighbor.h]: hmm, these two can never be on, can they?

doesn't matter much, but maybe make this an enum of its own?

```c
enum {
        NEIGHBOR_MAC,
        NEIGHBOR_IP,
        _NEIGHBOR_MAX,
        _NEIGHBOR_INVALID = -1,
};
```

File: src/network/networkd-neighbor.c (hunk 1 above)
Comment (yuwata) [reply src/network/networkd-neighbor.c]: Done.

File: src/network/networkd-neighbor.h (hunk 2 above)
Comment (yuwata) [reply src/network/networkd-neighbor.h]: Done.

**Comment by yuwata** [comment]
@poettering I've force-pushed a revised version. In the revised version, I applied the above two suggestions.

**Pull request merged**